registry/remote/repository.go

```diff
@@ -437,6 +437,59 @@ func filterReferrers(refs []artifactspec.Descriptor, artifactType string) []arti
 	return refs[:j]
 }
 
+// Extension provides information on extensions provided on a Registry.
+// Reference: https://github.com/opencontainers/distribution-spec/blob/main/extensions/_oci.md#extensions-property-descriptions
+type Extension struct {
+	Name        string   `json:"name"`
+	URL         string   `json:"url"`
+	Description *string  `json:"description,omitempty"`
+	Endpoints   []string `json:"endpoints"`
+}
+
+// DiscoverExtensions lists all supported extensions in current repository.
+// Reference: https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md#api-discovery
+func (r *Repository) DiscoverExtensions(ctx context.Context) ([]Extension, error) {
```

Comment:
Does this API support pagination?

Related: https://github.com/opencontainers/distribution-spec/tree/main/extensions#pagination